### ISSUE TITLE
place created host config in --prefix dir

### ISSUE THREAD

**Issue by cyrush** [LLNL/uberenv#44]
The used to be placed in the prefix dir, but now they placed in the dir where uberenv is run?
(I think we may have discussed this before)

I often use a few uberenv dirs -- I don't want the generated host configs to write over each other. 

Especially with the new strategy to stop at the hostconfig stage, b/c the install dir doesn't even exist -- if the host-config is written over, its gone. 

An option would be fine if folks are relying on this new behavior.

**Comment by white238** [comment]
To my knowledge isn't handled at the uberenv level.  It is handled by the individual package's spack package.py.  Though we should have a unified solution to this.  This PR would help with this:

https://github.com/spack/spack/pull/19316/

**Comment by cyrush** [comment]
with the old setup the host config was installed with the fake package for sure:

```
ls uberenv_libs/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/uberenv-conduit-0.0.0-bgxrwhrdhehspvsulco5blmxeyru4ug5/
elder.llnl.gov-darwin-mojave-x86_64-clang@10.0.0-apple-conduit.cmake	lib
```

Then it also ended up at:
```
uberenv_libs/
elder.llnl.gov-darwin-mojave-x86_64-clang@10.0.0-apple-conduit.cmake
```

I expect the host-config to be generated in the staged dir -- but that is now transient, how does it end up getting written above the prefix by the spack package?
Sounds like I am doing something wrong. I'll look into it more.

**Comment by cyrush** [comment]
For my case it was the fake package placing it:
`https://github.com/LLNL/conduit/blob/b3d1ba0feb41f4ed558a805a0023eef30b0485cf/scripts/uberenv/packages/uberenv-conduit/package.py#L79`

It's a bit scary to have the kind of logic in the proper spack package. I think uberenv should needs to help us here.

**Comment by cyrush** [comment]
Good news: I think the logic in the spack package is fine.

It has to do with where we tell the `dev-build` to execute.

I'll think of a knob that makes sense.

**Comment by cyrush** [comment]
```
        self.pkg_src_dir = os.path.join(self.uberenv_path,self.pkg_src_dir)
        if not os.path.isdir(self.pkg_src_dir):
            print("[ERROR: package_source_dir '{0}' does not exist]".format(self.pkg_src_dir))
            sys.exit(-1)
```

looks like `pkg_src_dir` is always eval-ed relative to the uberenv script itself, can we change this to be relative to the dest dir (aka prefix) ?

**Comment by cyrush** [comment]
dev build treats the passed source dir as the stage dir, so everything will always be written there. Played some tricks with symlinks, but that doesn't help.  A normal install with `stop-at-a-phase` would be the safest path, not sure if thats possible.

**Comment by adrienbernede** [comment]
@cyrush, I agree that the generated host-config files currently end up in the project dir.
The way I prevent them from overwriting one another is to name them with a short spec hash as suffix.
That's a good practice in general since it allows to differentiate very similar host-configs without putting a zillion parameters in the file name.
What do you think?

**Comment by cyrush** [comment]
@adrienbernede we can also use `-u {UNTIL}` with `spack install`.  I think that would address my concerns vs using `dev-build`

The only drawback is an extra fetch of source code -- I am going to explore this as an option.
